### ISSUE TITLE
Nethunter on top of Lineage 15.1 ?

### ISSUE THREAD

**Issue by noyb666** [offensive-security/kali-nethunter#1168]
### Device:

Nexus Angler 6p

### OS version (KitKat/Lollipop/Marshmallow/Nougat):

Oreo 8.1.0  OPM6.171019.030.E1 , Jul 2018

## Issue:
So I got a Nexus 6p and am happy with the device.

To get rid of Goolge I installed Lineage and that works fine too.

But my goal is to have Kali Nethunter installed to get a penetest lab.

My understanding of how that will look in the end is limited.

Can I just start to flash nethunter on this device with Lineage 15.1 installed?

I tend to think not, because in the install guide it says:

"You can download and install Kali Linux NetHunter images on the following Android devices:"

And then lists:

"Nexus 6 Marshmallow and Nexus 6 Lollipop"

(BTW: Lollipop is not supported anymore!)

Source: https://www.offensive-security.com/kali-linux-nethunter-do…/

To make it more confusing, on the github site it says:

"Supported Devices and ROMs:

Nexus 6P (angler) 6.0.1 or 7.1.1"

Confusing, because 6.0.1 is Marshmallow but 7.1.1 is Nougat and has not been mentioned before...

Source: https://github.com/offensive-security/kali-nethunter/wiki

So my questions: Can I flash it on top of a Lineage 15.1 ? If not, do I have to flash Lollipop or Nougat first and then Nethunter? 

How will the result look like, I mean: once I have Nethunter installed, can I EASILY switch to Android or even Lineage or will it be just for Nethunter with the need to start all over again if I want to use it as a "normal" phone?

In short: 

Actual state: Android Oreo 8.1.0,  OPM6.171019.030.E1 , Jul 2018, and twrp-3.2.2-0-angler are installed, and Lineage 15.1 on top of that.

Goal: To install kali-nethunter

Can somebody give me a lead how to proceed?

**Comment by 0E800** [comment]
With the amount of time you would save, you could afford 2 raspberry pi 3s.
Use a rpi3 for your Kali box or an old laptop running off USB drive.
Phones make for fickle and unpredictable pentesting devices .

**Comment by noyb666** [comment]
While that might be all true, it doesn't answer my question.

**Comment by 0E800** [comment]
The answer your question is yes.
Any kernel developer with some android experience and determination could
make it work.
You dont even know what it is or what to expect and also expect someone to
put the pieces together for you .
If your familiar with flashing ROMs and not worried about wiping data then
be bold and hack that isht.

This isn't a game that you get to the finish line in the first year playing.

Recommends:

Flash to whatever version is cookie cutted to your phone .You will need to
install whatever Rom matches prefab kernel.
If you can get all the nifty things to work then your good ... Or, you can
pick up android kernel development as a hobby and try compiling a kernel
for Lineage Oreo.

**Comment by noyb666** [comment]
"You dont even know what to expect" I admitted that already, that's why I'm asking.
"expect someone to put the pieces together for you ." No, I just don't want to loose precious time again because people can not write a proper install guide. 
Example: Trying to install Lineage I got errors all the time for not having the correct OPM, just because they foolishly didn't mention that it will work only with July 18 Oreo! 
Thankfully I found this guide and then it was easy: https://forum.xda-developers.com/nexus-6p/orig-development/rom-lineageos-15-0-nexus-6p-angler-t3716789

And the install guides for Nethunter are even more confusing, as I have mentioned before....

**Comment by noyb666** [comment]
And:
"Any kernel developer with some android experience and determination could
make it work."
I'm obviously none... :-) hence my question.

**Comment by 0E800** [comment]
I'm not trying to sell you anything sir. My time has no value. Glad you won
the day and it all turned out peaches.

**Comment by noyb666** [comment]
No worries, but I like to take it step by step.... once I'm done with this here, I might turn to your raspberry pi 3 idea, thanks for the input.

**Comment by DrillNBlast** [comment]
Because this endeavor is so device and software version dependent I found it best to avoid install guides from xda unless you intend to match hardware and software.

I spent weeks at the github source reading through scripts and readme's and on kali.org<http://kali.org> reading their docs before being able to wrap my head around the best way to approach this... If you haven't done so yet I suggest you do the same

Sent from my iPhone

On 21 Jul 2018, at 09:15 am, "noyb666" <notifications@github.com<mailto:notifications@github.com>> wrote:


"You dont even know what to expect" I admitted that already, that's why I'm asking.
"expect someone to put the pieces together for you ." No, I just don't want to loose precious time again because people can not write a proper install guide.
Example: Trying to install Lineage I got errors all the time for not having the correct OPM, just because they foolishly didn't mention that it will work only with July 18 Oreo!
Thankfully I found this guide and then it was easy: https://forum.xda-developers.com/nexus-6p/orig-development/rom-lineageos-15-0-nexus-6p-angler-t3716789

And the install guides for Nethunter are even more confusing, as I have mentioned before....

**Comment by DrillNBlast** [comment]
Oh and to answer an earlier question the end result is you have a working normal device with a new folder structure branch specific for kali that is meant to mimic its structure on a laptop or PC

Sent from my iPhone

On 21 Jul 2018, at 09:37 am, "noyb666" <notifications@github.com<mailto:notifications@github.com>> wrote:


No worries, but I like to take it step by step.... once I'm done with this here, I might turn to your raspberry pi 3 idea, thanks for the input.

**Comment by DrillNBlast** [comment]
And one more thing... If you search on GitHub enough you may find a kernal already compiled to your situation outside the devices listed on kali.org<http://kali.org> if you don't feel like getting into kernal development...

Sent from my iPhone

On 21 Jul 2018, at 09:37 am, "noyb666" <notifications@github.com<mailto:notifications@github.com>> wrote:


No worries, but I like to take it step by step.... once I'm done with this here, I might turn to your raspberry pi 3 idea, thanks for the input.

**Comment by noyb666** [comment]
"try compiling a kernel" or "I spent weeks"...
let's keep it real guys... I still have a life....

I asked a simple question, let's keep it simple and goal oriented:

1. Just following the install guide with the device in the state it is now as described above.... Do I have a chance it'll work? Anybody succeeded in doing so? Yes, or No.

2. If not: Downgrade to Nougat? If so which version? The latest available? 7.1.2 (N2G48C, Aug 2017)
Yes, or No.

I'm still frustrated the Lineage guys wasted my precious time in failing to outline that 15.1 works ONLY with the July Oreo....
Trying to avoid a deja vu...

**Comment by DrillNBlast** [comment]
[QUOTE] "No, I just don't want to loose precious time again because people can not write a proper install guide."

Quit wasting everyone else's precious time then if you aren't willing to put in your own. We know what it is you want but nobody is likely to help you with answers that you can easily find yourself if you bothered to try! The answers you were offered already actually filled in the blanks that are near impossible to cover in an install guide because this project is device and software version dependent.

You were offered advice and suggestions which is more than anyone had to do but you call it lecturing? Don't like it? Pay someone else to do it...

Sent from my iPhone

On 21 Jul 2018, at 07:27 pm, "noyb666" <notifications@github.com<mailto:notifications@github.com>> wrote:


WTF are you talking? My homework? If I'd want to go the trial and error way I wouldn't open this thread.

I asked a VERY SPECIFIC question... instead I got lectured about raspberry pi, compiling a kernel or spending weeks reading scripts....
That's a great way to learn a lot, no doubt, but not what I am looking for now.

So who is insulting whom here by going off topic ??

In short: if you don't know or don't want to answer my question, it's totally ok not to comment and leave the space for others who went through the same experience and are kind enough to share their knowledge.

**Comment by 0E800** [comment]
Not even good at feigning appreciation.
Very quick to accuse when ain't no else even batting a eye in your direction .

Lame ass smart phone generation thinking everything should be presented in some form or another as if everything is dependant on  competing for your attention.

You got character flaws that need to get dealt with before you die and get reincarnated no better off than you are now.

Upgrade your gray matter and humble your attitude plebian.

**Comment by DrillNBlast** [comment]
lol I was trying not to be so harsh but I don't disagree

Sent from my iPhone

On 21 Jul 2018, at 08:53 pm, "0E800" <notifications@github.com<mailto:notifications@github.com>> wrote:


Not even good at feigning appreciation.
Very quick to accuse when ain't no else even batting a eye in your direction .

Lame ass smart phone generation thinking everything should be presented in some form or another as if everything is dependant on competing for your attention.

You got character flaws that need to get dealt with before you die and get reincarnated no better off than you are now.

Upgrade your gray matter and humble your attitude plebian.

**Comment by 0E800** [comment]
Not working with feelings.
Being rude is not glamorous however there is such a thing as being righteously rude for a reason.

**Comment by noyb666** [comment]
Lol, this is even more ridiculous than Facebook...

Going off topic and getting personal, even a bigger waste of time.

As I said: If you don't have the experience or answer I'm looking for, it's totally ok to NOT COMMENT and go on with your life, if you have any....

**Comment by 0E800** [comment]
.oO(Is thinking out loud that your a douche-canary still make it look like
I'm trying to have the last word?)

I am not.

But feel free to reply, I promise I wont reply back this time.

**Comment by noyb666** [comment]
"I promise I wont reply back this time."
Great.
Next....

**Comment by mkgrage** [comment]
@noyb666 Wow, its so impolite manner. Its a free discussion and everyone should feel respect each other.

**Comment by mkgrage** [comment]
@noyb666 You have to know, I was on your position few times ago....I run my device using RR 6.1 Oreo 8.1 and at the same time I took a great interest regarding nethunter. Its a bunch questions because:
1. My device is an infamous gadget (lentislte), so only a little Rom and discussion still exist till this time (thanks to Corinna for her great work)
2. I have had run my device using Oreo, which no forum discuss regarding nethunter on oreo.

And you know what? He is I have done:

1. Understanding what is nethunter, and how it can be installed on some device running different ROM.
(Theres is a bundh even on xda explaining how to install nethunter on any device, yes on any device and any OS version)
2. Nethunter consist of chroot and kernel, which chroot you can download it from nethunter website, and it can be appied to any device and any OS version (CMIIW). But regarding kernel, this is the most important to nethunter running well on your device. 
3. In order to get nethunter supported-kernel, you can search it (in my case-its no one, so I need to compile it by my self. I'm a noob regarding linux etc, but it make me very interest to learn since A).
4. To compile your kernel, go to nethunter github, youtube, xda, and DISCUSSION with POLITE MANNER, RESPECT EACH OTHER.

And the conclusion is, yes I got my nethunter running well on my device with RR 6.1 Oreo ROM.

**Comment by noyb666** [comment]
You know, the reason why I got the Nexus Angler is exactly, because it IS supported and I don't know how to and don't want to fool around with the kernel. 
But now I see, that already this is far from easy, had trouble installing Lineage and there's not even consensus what Android you shall use. So all I asked is: does it make sense, work with 8.1 or do I have to downgrade to Nougat ?
I also found here: https://build.nethunter.com/nightly/2017.11-18-1618/
a "kernel-nethunter-angler-oreo-2017.11-18-1618.zip" but honestly don't even understand if that's what I'm looking for and if I can just flash it now on the phone. And that's exactly why I have opened this thread, hoping that some more experienced guys who went through this can give advice.

**Comment by mkgrage** [comment]
@noyb666 Did you read my last statements? Its YES, you can use nethunter on 8.1. 
Then, lets start from what you have, what have you done.

**Comment by DrillNBlast** [comment]
@KiraRyu you are much kinder than I am to stick to helping... I just hope you get more appreciation for your efforts than anyone else did

Sent from my iPhone

On 23 Jul 2018, at 09:50 am, "KiraRyu" <notifications@github.com<mailto:notifications@github.com>> wrote:


@noyb666<https://github.com/noyb666> You have to know, I was on your position few times ago....I run my device using RR 6.1 Oreo 8.1 and at the same time I took a great interest regarding nethunter. Its a bunch questions because:

  1.  My device is an infamous gadget (lentislte), so only a little Rom and discussion still exist till this time (thanks to Corinna for her great work)
  2.  I have had run my device using Oreo, which no forum discuss regarding nethunter on oreo.

And you know what? He is I have done:

  1.  Understanding what is nethunter, and how it can be installed on some device running different ROM.
(Theres is a bundh even on xda explaining how to install nethunter on any device, yes on any device and any OS version)
  2.  Nethunter consist of chroot and kernel, which chroot you can download it from nethunter website, and it can be appied to any device and any OS version (CMIIW). But regarding kernel, this is the most important to nethunter running well on your device.
  3.  In order to get nethunter supported-kernel, you can search it (in my case-its no one, so I need to compile it by my self. I'm a noob regarding linux etc, but it make me very interest to learn since A).
  4.  To compile your kernel, go to nethunter github, youtube, xda, and DISCUSSION with POLITE MANNER, RESPECT EACH OTHER.

And the conclusion is, yes I got my nethunter running well on my device with RR 6.1 Oreo ROM.

**Comment by Almaincognito** [comment]
The guys that attacked noyb666 were ridiculous liars, and to a remarkable extent. Im not including people who answered his question with as much info as they possessed and were simply sarcastic or (deliberately) obfuscating.  That shit was still wrong, but obfuscating is how we getya.

The person who said to build a raspberry pi 3 gave good advice as a pentest solution, but I feel they did so either in ignorance of what the 5 and 6p can do or, more likely, don't use arm devices for pentests. In any case, a) The pi 3 doesn't merely lack the internal adapter allowing for packet injection/monitor mode (yes, wlan0, as opposed to the external adapter necessary for a pi 3 to, oh wait, I forgot, with only the limited ports available, going wlan1 up means you have to sacrifice something else.  B) Even if you build a modern pi with the proper broadcom chip, you'd still not be able to attack with HID devices, and saving those from our pool means that there's not much to recommend it except for extremely limited red team cases. If you use the latest pwnpi devices and combine it with Hak5 cloud based options (and latest payloads for ducky and bunny scripts..) AND.... Nevermind.


The person who was yelling at the kid to look at xda, do his own work (SAYS THE GITHUB WTF) without any hints, and, as the final coup deimbecile, to brag using the pronoun we saying that we did all the work stop being lazy is some next level Carlos Mencia shit.

Shame on you, kiddo.

**Comment by 0E800** [comment]
I disagree with your perception. Don't like how you vaguely include everyone in this thread that wasted time sincerely intending to help OP, and tag us as liars. 

That you think rpi3 makes for a poor choice for a mobile arsenal drop box because it has no packet injection capabilities is adolescent as well. 

Perimeter hacking is not the essence of a pentest. Newbs are attracted to nethunter mostly because they want to hack WiFi. 

We are weaponizing IOT and at the same time offering or trying to help others with the pitfalls we had to overcome as a way of playing it forward. 

Having decent table manners and a healthy serving of patience is really all you need to gain favor from the overnight developers that still haunt these parts.

OP had a bad attitude from the gate. You want to stick your neck out for this dude then by all means... paint yourself as such.
